agent/api/container.go

```diff
@@ -565,3 +563,26 @@ func (c *Container) BuildContainerDependency(contName string,
 	deps.ContainerDependencies = append(deps.ContainerDependencies, contDep)
 	c.TransitionDependenciesMap[dependentStatus] = deps
 }
+
+// BuildResourceDependency adds a new resource dependency and status that satisfies
+// the dependency
+func (c *Container) BuildResourceDependency(resourceName string,
```

Comment:
Can you please document what each of these fields mean, with an example as well?